File: services/payment-orchestrator-java/src/main/java/com/acme/payments/orchestrator/Application.java
package com.acme.payments.orchestrator;

import com.acme.payments.orchestrator.application.OutboxEventPublisher;
import com.acme.payments.orchestrator.application.PaymentOrchestratorService;
import com.acme.payments.orchestrator.application.IdempotencyStore;
import com.acme.payments.orchestrator.application.PaymentRepository;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.scheduling.annotation.EnableScheduling;
import org.springframework.context.annotation.Bean;

@SpringBootApplication
@EnableScheduling
public class Application {
    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }

    @Bean
    PaymentOrchestratorService paymentOrchestratorService(
            PaymentRepository paymentRepository,
            IdempotencyStore idempotencyStore,
            OutboxEventPublisher outboxEventPublisher
    ) {
        return new PaymentOrchestratorService(paymentRepository, idempotencyStore, outboxEventPublisher);
    }
}

File: services/pix-boleto-kotlin/src/main/kotlin/com/acme/payments/pixboleto/Application.kt
package com.acme.payments.pixboleto

import com.acme.payments.pixboleto.provider.AsaasAdapter
import com.acme.payments.pixboleto.provider.IuguAdapter
import com.acme.payments.pixboleto.provider.MercadoPagoAdapter
import com.acme.payments.pixboleto.provider.PagBankAdapter
import com.acme.payments.pixboleto.provider.PaymentProviderAdapter
import com.acme.payments.pixboleto.provider.ProviderRouter
import com.acme.payments.pixboleto.provider.StripeAdapter
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean

@SpringBootApplication
class Application {
    @Bean
    fun asaasAdapter(): PaymentProviderAdapter = AsaasAdapter()

    @Bean
    fun mercadoPagoAdapter(): PaymentProviderAdapter = MercadoPagoAdapter()

    @Bean
    fun pagBankAdapter(): PaymentProviderAdapter = PagBankAdapter()

    @Bean
    fun iuguAdapter(): PaymentProviderAdapter = IuguAdapter()

    @Bean
    fun stripeAdapter(): PaymentProviderAdapter = StripeAdapter()

    @Bean
    fun providerRouter(adapters: List<PaymentProviderAdapter>) = ProviderRouter(adapters)
}

fun main(args: Array<String>) {
    runApplication<Application>(*args)
}
